lib/internal/streams/writable.js

```diff
@@ -934,3 +940,12 @@ Writable.fromWeb = function(writableStream, options) {
 Writable.toWeb = function(streamWritable) {
   return lazyWebStreams().newWritableStreamFromStreamWritable(streamWritable);
 };
+
+Writable.prototype[SymbolAsyncDispose] = function() {
+  let error;
+  if (!this.destroyed) {
+    error = this.writableFinished ? null : new AbortError();
+    this.destroy(error);
+  }
+  return new Promise((resolve, reject) => eos(this, (err) => (err && err !== error ? reject(err) : resolve(null))));
```

Comment:
```suggestion
  return new Promise((resolve, reject) => eos(this, (err) => (err && err.name !== 'AbortError ? reject(err) : resolve(null))));
```

I think this is a better variation.